docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/EntityFrameworkEnumerable.cs

```diff
@@ -0,0 +1,128 @@
+        public float PercentResidental { get; set; }
+        public float PercentNonRetail { get; set; }
+        public float CharlesRiver { get; set; }
+        public float NitricOxides { get; set; }
+        public float RoomsPerDwelling { get; set; }
+        public float PercentPre40s { get; set; }
+        public float EmploymentDistance { get; set; }
+        public float HighwayDistance { get; set; }
+        public float TaxRate { get; set; }
+        public float TeacherRatio { get; set; }
+    }
+
+    public partial class MLNetExampleContext : DbContext
+    {
+        public DbSet<HousingData> Housing { get; set; }
+
+        public MLNetExampleContext(DbContextOptions<MLNetExampleContext> options)
+            : base(options)
+        {
+        }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+                optionsBuilder.UseSqlServer("Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory;Trusted_Connection=True;ConnectRetryCount=0");
+            }
+        }
+    }
+
+    public static class EntityFrameworkEnumerable
+    {
+        private static IEnumerable<HousingData> housingData;
+
+        public static void Example()
+        {
+            // Download the housing data set and get the file name.
+            var housingFile = SamplesUtils.DatasetUtils.DownloadHousingRegressionDataset();
+
+            // Read data from file into an IEnumerable.
+            var data = File.ReadAllLines(housingFile)
+                .Skip(1) // Skip the header row
+                .Select(l => l.Split('\t'))
+                .Select(i => new HousingData
+                {
+                    MedianHomeValue = float.Parse(i[0]),
+                    CrimesPerCapita = float.Parse(i[1]),
+                    PercentResidental = float.Parse(i[2]),
+                    PercentNonRetail = float.Parse(i[3]),
+                    CharlesRiver = float.Parse(i[4]),
+                    NitricOxides = float.Parse(i[5]),
+                    RoomsPerDwelling = float.Parse(i[6]),
+                    PercentPre40s = float.Parse(i[7])
+                });
+
+            // Options to pass to the DbContext to use an in-memory database for
+            // Entity Framework and to give the database a name.
+            var options = new DbContextOptionsBuilder<MLNetExampleContext>()
+                .UseInMemoryDatabase(databaseName: "TestData")
+                .Options;
+
+            // Add the data to the in-memory database via Entity Framework.
+            using (var dbContext = new MLNetExampleContext(options))
+            {
+                foreach (var dataItem in data)
+                {
+                    dbContext.Add(dataItem);
+                }
+
+                var rowsAffected = dbContext.SaveChanges();
+
+                housingData = dbContext.Housing.ToList();
```

Comment:
This ToList() and the ReadAllLines() code that initially loads the data from a text file require that all the data fits in memory at the same time (twice actually). As @singlis mentioned, one of the goals with had in mind was to show how to avoid this.